lib/tutorials/en_US/routing.md

```diff
@@ -110,14 +112,57 @@ There are two query parameters here, `name=ferris` and `location=chicago`.  In h
 server.route({
     method: 'GET',
     path: '/',
-    handler: (request, h) => {
+    handler: function (request, h) {
 
         return `Hello ${request.query.name}!`;
     }
 });
 ```
 Here, we simply access the `name` query parameter and return it in the handler, which would read `Hello ferris!`.
 
+For more complex query structures, you may opt to use the `qs` module.  Consider the following:
+
+```js
+server.route({
+    method: 'GET',
+    path: '/',
+    handler: function (request, h) {
+
+        return request.params;
+    }
+});
+```
+If you sent the request `localhost:3000?foo[bar]=baz`, hapi, by default would return `{ "foo[bar]": "baz" }`.  
+
+With the `qs` module, we can parse the query string out.  An example: 
+
+```js
+const QS = require('qs');
+
+server.route({
+    method: 'GET',
+    path: '/',
+    handler: function (request, h) {
+
+        const query = QS.parse(request.query);
```

Comment:
Use the route query parser option, not manually.

Reply:
here is the API link: https://hapijs.com/api#server.options.query.parser

Reply:
Thanks...took a little digging, but was able to find it.